### PR TITLE
release: copy cloudflare assets when promoting

### PR DESCRIPTION
**Pull request by MoLow**
While I believe assets can be copied directly from Jenkins in many cases, I think it is better to start with this approach where everything is promoted from within the www server.

this still doesn't handle the latest soft links. see discussion https://github.com/nodejs/release-cloudflare-worker/issues/33

### CODE REVIEW

File: ansible/www-standalone/tools/promote/upload_to_cloudflare.sh
Comment (richardlau) [inline ansible/www-standalone/tools/promote/upload_to_cloudflare.sh:1]: If this file needs things to be set from the caller (e.g. `$dist_rootdir`) then it should probably check and error if those are not set.

File: ansible/www-standalone/tools/promote/promote_release.sh [hunk 1]
```diff
@@ -37,6 +37,8 @@ while true; do
   if [ "X${yorn}" == "Xy" ]; then
       . ${__dirname}/_promote.sh $site $2
       nodejs-latest-linker /home/dist/${site}/release/ /home/dist/${site}/docs/
+      dstdir=/home/dist/${site}/docs
+      . ${__dirname}/upload_to_cloudflare.sh $site $2
```
Comment (richardlau) [inline ansible/www-standalone/tools/promote/promote_release.sh:41]: Why is this needed for releases but not the non-release (e.g. nightlies, v8-canary) builds?

**Comment by richardlau** [comment]
> While I realize we do not run the playbook for the www server against the live www server, for my own benefit I'm assuming this needs something similar to #3501 to get the aws client and secrets onto the server?

Actually we probably need to make sure we're okay giving the `dist` user access to the secrets to publish to R2 storage as that is who the releasers log into the www machine as to do the promotion step. 
Refs: https://github.com/nodejs/node/blob/7b624c30b2902375c3c81885fc964b81da0891db/tools/release.sh#L14

**Comment by MoLow** [comment]
> Actually we probably need to make sure we're okay giving the dist user access to the secrets to publish to R2 storage as that is who the releasers log into the www machine as to do the promotion step.
Refs: [nodejs/node@7b624c3/tools/release.sh#L14](https://github.com/nodejs/node/blob/7b624c30b2902375c3c81885fc964b81da0891db/tools/release.sh#L14)

Why would that not be ok? these credentials can only access this r2 bucket and nothing else

**Comment by MoLow** [comment]
> While I realize we do not run the playbook for the www server against the live www server, for my own benefit I'm assuming this needs something similar to https://github.com/nodejs/build/pull/3501 to get the aws client and secrets onto the server?

I already uploaded that manually for the initial file sync, but added to the code for the sake of documentation

File: ansible/www-standalone/tools/promote/promote_release.sh (hunk 1 above)
Comment (MoLow) [reply ansible/www-standalone/tools/promote/promote_release.sh:41]: this line will actually become redundant once https://github.com/nodejs/release-cloudflare-worker/pull/39 is merged

**Comment by MoLow** [comment]
I'v deployed all files manually besides `ansible/www-standalone/tools/promote/resha_release.sh`, manually re-created `.done` files for the last nightly build and promotion was performed to R2: https://r2.nodejs.org/download/nightly/v21.0.0-nightly20231001092fb9f541/

**Comment by MoLow** [comment]
https://r2.nodejs.org/download/nightly/v21.0.0-nightly2023100285c09f178c/ was deployed, 100% automatically now

**Review by richardlau — APPROVED** [review]

**Comment by richardlau** [comment]
> > Actually we probably need to make sure we're okay giving the dist user access to the secrets to publish to R2 storage as that is who the releasers log into the www machine as to do the promotion step.
> > Refs: [nodejs/node@7b624c3/tools/release.sh#L14](https://github.com/nodejs/node/blob/7b624c30b2902375c3c81885fc964b81da0891db/tools/release.sh#L14)
> 
> Why would that not be ok? these credentials can only access this r2 bucket and nothing else

I may be being a tad cautious but most releasers (who have access to `dist` and run the promotion script to promote releases) are not members of build. We currently prevent releasers from changing existing downloads after one week (see https://github.com/nodejs/build/pull/2020) -- I've no idea if that's possible with R2.
https://github.com/nodejs/build/blob/726a7ca3a4724b3de72083ae43118f0317fc7d71/ansible/www-standalone/resources/scripts/dist-perms#L1-L10

While we do generally trust releasers we have had one incident recently where a releaser lost their laptop which meant we had to revoke their ssh keys to `dist`. I guess maybe it'd be okay if we're able to rotate the secret used to publish to R2 in the future if we had a similar potential compromise.

**Comment by MoLow** [comment]
updated files manually
